apps/cli-daemon/src/app/generators/generators.service.ts

```diff
@@ -1,26 +1,67 @@
 import { spawnSync, SpawnSyncOptionsWithBufferEncoding } from 'child_process';
+import { readFileSync } from 'fs';
 import { resolve as pathResolve } from 'path';
 
-import { Injectable } from '@nestjs/common';
+import { Injectable, Logger } from '@nestjs/common';
 
 import { GENERATE_COMMAND } from '../ng-commands';
 import { SessionService } from '../session/session.service';
 
 import { ExecResult } from './dto';
+import { GeneratorDefinition, Schema } from './generators.interface';
+import {
+  convertKeyToArgument,
+  getGeneratorDefinition,
+  getGeneratorNames,
+} from './utils';
 
 @Injectable()
 export class GeneratorsService {
+  private readonly logger = new Logger(GeneratorsService.name);
+
   constructor(private readonly sessionService: SessionService) {}
 
-  execSync(args: string[] = [], ng = GENERATE_COMMAND): ExecResult {
+  execSync(params: Record<string, any>, ng = GENERATE_COMMAND): ExecResult {
+    const args = this.getArgsFromParams(params);
     const options: SpawnSyncOptionsWithBufferEncoding = {
       cwd: pathResolve(this.sessionService.cwd),
       env: process.env,
       shell: true,
     };
     const command = [ng].concat(args || []).join(' ');
+    this.logger.log(`Running ${command} for ${options.cwd}`);
     const res = spawnSync(command, options);
 
     return new ExecResult(res);
   }
+
+  getAllGenerators(): GeneratorDefinition[] {
+    const schemasPath = pathResolve(
+      this.sessionService.cwd,
+      'node_modules/@schematics/angular'
```

Comment:
It can't be, this.sessionService.cwd it's a getter to get the connected workspace path, we can't save this base path outside because we can't listen to changes so we need to redefine on every function run

Reply:
I dont understand what it can't be a const...
```
 const schemasPath = pathResolve(
      this.sessionService.cwd,
      SCHEMATICS_PATH
      )
```